### ISSUE TITLE
[Tune][Docs] Torch Lightning integration reports _last_ metric, not best when using early stopping

### ISSUE THREAD

**Issue by samgoldman97** [ray-project/ray#28734]
### Description

When having ray tune monitor the validation loss, my expectation was that Ray Tune would use the _best_ validation loss from the given trial. However, it uses the _last_  value reported, even when using Torch Lightning's early stopping callback. Because of this, ray tune can be sensitive to the patience setting in torch lightning's early stopping callback (I.e., with high patience, torch lightning continues training well beyond the best observed validation loss, shuffling the order of trials).

Additional documentation could help, or modifying the behavior of torch lightning's integration directly to keep a max/min of each reported metric. 

### Link

https://docs.ray.io/en/latest/_modules/ray/tune/integration/pytorch_lightning.html
